test/unit_test.c

```diff
@@ -4141,6 +4141,14 @@ int main(void) {
   if (s_abort != 0) return EXIT_FAILURE;
 #endif
 
+
+#if 0
```

Comment:
This is incorrect, we have to call mbedtls_psa_crypto_free(). Tests are not failing because library versions involved do not require that function call. I explicitly added it in #3281 after investigating a report in #3278 with a specific build of the library
Unfortunately, I didn't realize that was using global data...